octree/include/pcl/octree/impl/octree_iterator.hpp

```diff
@@ -281,6 +281,56 @@ namespace pcl
 
       return (*this);
     }
+
+    //////////////////////////////////////////////////////////////////////////////////////////////
+    template<typename OctreeT>
+    OctreeBreadthFirstAtDepthIterator<OctreeT>::OctreeBreadthFirstAtDepthIterator (unsigned int depth_arg) :
+        OctreeBreadthFirstIterator<OctreeT> (depth_arg), depth_(depth_arg)
+    {
+      OctreeBreadthFirstIterator<OctreeT>::reset ();
+
+      // initialize iterator
+      this->reset ();
+    }
+
+    //////////////////////////////////////////////////////////////////////////////////////////////
+    template<typename OctreeT>
+    OctreeBreadthFirstAtDepthIterator<OctreeT>::OctreeBreadthFirstAtDepthIterator (OctreeT* octree_arg, unsigned int depth_arg) :
+        OctreeBreadthFirstIterator<OctreeT> (octree_arg, depth_arg), depth_(depth_arg)
+    {
+      OctreeBreadthFirstIterator<OctreeT>::reset ();
+
+      // initialize iterator
+      this->reset ();
+    }
+
+    //////////////////////////////////////////////////////////////////////////////////////////////
+    template<typename OctreeT>
+    OctreeBreadthFirstAtDepthIterator<OctreeT>::~OctreeBreadthFirstAtDepthIterator ()
+    {
+    }
+
+    //////////////////////////////////////////////////////////////////////////////////////////////
+    template<typename OctreeT>
+    void OctreeBreadthFirstAtDepthIterator<OctreeT>::reset ()
+    {
+      OctreeBreadthFirstIterator<OctreeT>::reset ();
+
+      // Set the current_state_ to the right depth
+      while( this->current_state_ && ( this->getCurrentOctreeDepth () != depth_ ) ) {
```

Comment:
Missing space after `while`, extra spaces within braces. `{` should go to the new line